### DISCUSSION TITLE
[7.0-beta] Multiple deployments to same host?

### DISCUSSION THREAD

**Discussion by YetiCGN** [deployphp/deployer#2557] [General]
- Deployer version:  7.0.0-beta.23
- Deployment OS: Ubuntu 20.04.2 LTS

How do I define two deployments onto the same host in Deployer 7?

With Deployer 5 and 6, we had this working:

    // Hosts
    host('stage')
        ->hostname('mytarget.example.org')
        ->user('deploy-stage')
        ->stage('stage')
        ->set('deploy_path', '/home/deploy-stage/www')
        ->set('branch', 'develop')
    ;
    // Hosts
    host('prod')
        ->hostname('mytarget.example.org')
        ->user('deploy-prod')
        ->stage('prod')
        ->set('deploy_path', '/home/deploy-prod/www')
        ->set('branch', 'develop')
    ;

I tried to do it like this:

    // Hosts
    host('mytarget.example.org')
        ->set('user', 'deploy-stage')
        ->set('stage', 'stage')
        ->set('deploy_path', '/home/deploy-stage/www')
        ->set('branch', 'develop')
    ;
    host('mytarget.example.org')
        ->set('user', 'deploy-prod')
        ->set('stage', 'prod')
        ->set('deploy_path', '/home/deploy-prod/www')
        ->set('branch', 'master')
    ;

which fails (obviously) with the following error:

      Host "mytarget.example.org" already exists.
      If you want to override configuration options, get host with getHost function.

          getHost('mytarget.example.org');

Unfortunately, 7.0.0-beta is the only version right now compatible with PHP 8.0, which we use for this new project.

**Comment by antonmedv** [comment]
Try this:
```php
host('domain.com/green', 'domain.com/blue')
    ->set('deploy_path', '~/{{hostname}}')
    ...
```
```
~
└── domain.com
    ├── green
    │   └── ...
    └── blue
        └── ...
```

Or this:
```php
host('production')
    ->hostname('domain.com')
    ->set('deploy_path', '~/domain.com');
    
host('beta')
    ->hostname('domain.com')
    ->set('deploy_path', '~/beta.domain.com');    
```

**Reply by YetiCGN** [reply]
Almost, since `hostname` was renamed to `setHostname` in 7. But this works like a charm, thanks!
